src/js/entry.js

```diff
@@ -26,20 +29,27 @@ export function chromeInit(libjwt) {
     const { store, middlewareListener, actions } = spinUpStore();
 
     // public API actions
-    const { identifyApp, appNav, appNavClick, clearActive } = actions;
-    libjwt.initPromise.then(() => {
+    const { identifyApp, appNav, appNavClick, clearActive, chromeNavUpdate } = actions;
+
+    const jwtAndNavResolver = libjwt.initPromise.then(() => {
```

Comment:
Looking at the function you gotta rewrite it because you will get race conditions. It would be better to use async/await but I am not sure if we can use it.
```JS
const jwtAndNavResolver = libjwt.initPromise
.then(() => libjwt.jwt.getUserInfo.then()) // you have to wait for getUserInfo to resolve here
.then((user) => {
  actions.userLogIn(user);
  loadChrome(user);
})
.then(() => sourceOfTruth(ibjwt.jwt.getEncodedToken).then()) // sourceOfTruth is Promise as well
.then(({ data }) => chromeNavUpdate(loadNav(data)))
.catch(() => {
  if (allowUnauthed()) {
    loadChrome(false);
  }
});
```